### DISCUSSION TITLE
Imputation

### DISCUSSION THREAD

**Discussion by beantkapoor786** [maize-genetics/phg_v2#189] [Q&A]
Hello, I have built a PHG pangenome with 6 diploid plant species (12 haplotypes) using default settings. I have Illumina sequences (~40x) coverage of a hybrid between two of my species in the pangenome. That hybrid is a cross between australis (primary and alternate haps) and wilking (primary and alternate haps). I used `find-paths` function to impute the sample using following command - 
```
/pickett_flora/software/phg/bin/phg find-paths         \
--path-keyfile data/key_files/path_finding_novogene_2024.txt         \
--hvcf-dir output/vcf_files/         \
--reference-genome output/updated_assemblies/fortune_primary.fa         \
--min-reads=1  \
--path-type diploid         \
--output-dir output/vcf_files_novogene_2024/         \
--use-likely-ancestors=true         \
--likely-ancestor-file=likely_ancestors.txt        \
--max-ancestors=4 
```

and here are the number of haplotype sequences imputed from each parental haplotype -

```
  16286  australis_alternate
  15589  australis_primary
      1  fallglo_alternate
      1  fortune_alternate
    650  inodora_alternate
   4419  inodora_primary
  31212  wilking_alternate
```
I have a couple of questions regarding this result - 
1. Why is PHG not able to impute any haplotype sequence coming out of wilking primary haplotype?
2. Why is it imputing paths from ancestors such as inodora_primary, inodora_alternate etc.?

Any help is much appreciated. Thank you!

**Comment by zrm22** [comment]
Hello,

For 1. I would double check the wilking_primary hvcf file's haplotypes to see if it has unique hapIds.  I think your intuition about it should be spread between primary and alternate similar to how it is from australis.  

For 2, the PHG path finding will often get off path in hard to map regions of the genome.  From some simple math you are about 7% off path which does seem consistent with some of our internal tests.   What are the contents of the likely_ancestors.txt file.  This should be a record of what the PHG is thinking is your most probable ancestors based on the read mappings.

**Comment by beantkapoor786** [comment]
Thank you for the response. 

1. I double checked the wilking primary hvcf file (generated after build and load module) and it has 37,576 unique hap ids. You are right about my intuition that it should be spread between wilking primary and alternate just as for australis.
2. Here are the contents of likely_ancestors.txt file for this sample -
```
sample	ancestor	gameteId	reads	coverage
F1_254_01	australis_primary	0	20883959	0.23917845087258216
F1_254_01	wilking_alternate	0	17034228	0.4342669522841375
F1_254_01	australis_alternate	0	9539013	0.5435147415655387
F1_254_01	inodora_primary	0	9259865	0.6495655227411407
```

**Reply by beantkapoor786** [reply]
Hello, was just wondering if you have any suggestions. Thank you.

**Reply by zrm22** [reply]
Hello,  We are currently investigating a bug in the imputation code which is causing some very poor imputation results.  

Once this is done we will do a new release then I would recommend building a new index and rerunning your samples to see if that improves the results.

**Reply by beantkapoor786** [reply]
Hi, I see a new version is released 2 hours ago. Does this one have the imputation bug fixed?

**Reply by zrm22** [reply]
No we are still working on it.  We likely will bump it a minor version and I will make an announcement on the Discussions when this is fixed.  I will also update this thread as well.
